### PR TITLE
[READY] Gold Slime Mob Reworks

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR reworks the pool the gold slime core reactions pulls from and the mobs therein.  The idea of this is to allow mobs made sentient from the cores to have actual benefits to the station instead of being monsters who can only smash things unless they're one of the very few animals that were unique.  This mostly involves itself in list removals and giving many of the mobs included in the spawns some unique abilities that can (mostly) help out the station.  The changes are as follows:

GOLDGRUB:
- Added to the spawn pool
- Can eat ore to store it within itself
- Has two abilities, Spit Ore and Burrow
- Spit Ore is simple, as it spits out all your currently consumed ores.
- Burrow allows the Gold Grub to bury itself under the surface from mining turfs (think asteroid ground and basalt ground from lavaland).  It can only come back out by utilizing one of these surfaces as well.
Intended Department: MINING

BASILISK:
- Can now drink lava.  Doing so heats the Basilisk up, giving it some temporary buffs:
  - Completely heals all damage when activated
  - Increases the Basilisk's speed to 0, from 3.
  - Changes the Basilisk's projectile to deal as much damage as an unupgraded KA.
  - Gives it that sick-looking alert sprite previously unusable by sentient basilisks
- These buffs run out after 5 minutes.  You cannot drink lava again until you've cooled down.
Intended Department: MINING

HIVEBOT:
- Removes the 4 duplicate hivebot types and the heavy hivebot from the pool, replacing them with a new mob called the hivebot mechanic.
- The hivebot mechanic can repair machinery and hivebots after a delay.
- The hivebot mechanic can also place down foam walls (the ones from foam grenades) to patch up possible breaches in the station's hull
Intended Department: ENGINEERING

FESTIVUS POLE:
- No longer takes damage in low O2 environments
- Nerfed health from 250 to 200
- Rubbing it (help intent) creates static charge, recharging power cells on the ground, in cyborgs, and in APCS by 75 in a radius of 2.
Intended Department: ENGINEERING/MISC

CRATE MIMIC:
- Both versions (yes, there were two identical crates in the pool) of mimic were removed and replaced with a crate mimic specifically built for xenobio
- Interacting with the crate on help intent will open and close it
- Can store things inside it like a real crate, including people
- Can ride on rideable mobs for ease of transport
- Crate can toggle a lock on itself, preventing anything from opening it or closing it.
Intended Department: CARGO/MISC

EYEBALL (You probably forgot this thing existed, didn't you?):
- Refactored to not stem from the carp mobtype.
- Can see through walls
- Night vision
- Still susceptible to punches
Intended Department: SECURITY

BEAR:
- All bears but the armored one removed from the pool (Terrygold is now chef-exclusive)
- Sentient bears can now be ridden
- Bear speed is now 0, making them slightly faster when player-controlled
- Can be combined with crate mimic to play Death Stranding as a bear
Intended Department:  MISC(Anyone could use a bear mount)

SPACE SHARK:
- Sentient ones are now also rideable
Intended Department: MISC(Anyone could also use a shark mount)

CREATURE/WIZARD'S PET GREEN BLOB THING:
- Has the ability to teleport around when not seen
- Can teleport to any location currently not being observed
- Health nerfed from 80 to 50
Intended Department: MISC

LIGHTGEIST:
- Can no longer harm people or break anything under any circumstance
- Healing power is now scaled with damage boosting methods (fugu).  An example of how this works would be that a lightgeist heals for 5 damage normally, but a fugu lightgeist now heals for 10 damage.

UPDATED:
- Goldgrub (Added)
- Basilisk
- Hivebot (New Mechanic Type)
- Festivus Pole
- Crate Mimic (New Xenobio Type)
- Eyeball
- Bear
- Space Shark
- Creature
- Lightgeist

STILL INCLUDED:
- Magicarp
- Chaos Magicarp
- Mi-go
- Wumborian Fugu
- Lusty Xenomorph Maid
- Pine Tree
- Blobbernaut
- Red-eyed Spider
- Faithless
- Viscerator
- Space Carp
- Dark Wizard
- Blob Spore
- Goose

REMOVED:
- All 3 Watchers
- 2 crate mimic duplicates
- All bears but Armored Bear (Terrygold goes home to Chef)
- Plasma Miners (Normal and Drill)
- Skeletons (Normal and Ice)
- Purple-Eyed Spider
- All 4 Hivebot Duplicates (Barely any difference between them)
- Heavy Hivebot
- Bee And Bee Queen (Moved to Water Reaction)
- Frogs (Moved to Water Reaction)
- Evil Crab (Moved to Water Reaction)

Other notes:
- Spinning to throw off riders now also applies to simple mobs.  Code was adjusted to accomodate for this change.

## Why It's Good For The Game

A majority of the gold slime pool has been a cesspool of unused mobs, most of which could only smash things, for way too long now.  These changes push a heft of the useless mobs into more station-assistive roles so those who take these ghost roles don't have to sit on their hands the whole time since they're useless outside of smashing things.

## Changelog
:cl: Indie-ana Jones
tweak: Xenobiologists have noted a change in results from the gold slime core's reaction.  Creatures previously thought to be useless are also notably more unique than before.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
> All bears but the armored one removed from the pool (Terrygold is now chef-exclusive)

Can you not

No, seriously, one of the fun things about gold core was some of the stupid shit you could get from it. I'm fine with some of these changes, but add crabs and the bear subtypes back in.

**Comment by IndieanaJones** [comment]
> 
> 
> > All bears but the armored one removed from the pool (Terrygold is now chef-exclusive)
> 
> Can you not
> 
> No, seriously, one of the fun things about gold core was some of the stupid shit you could get from it. I'm fine with some of these changes, but add crabs and the bear subtypes back in.

I'd be okay with adding the other bears back in, since the rideable things applies to them too.  However, Terrygold is sort of meant to be a Chef-exclusive mob, I believe his inclusion in Xenobio was a mistake from the start.

Also, why do you want Evil Crabs back in?  They're not even capable of doing anything.

**Comment by Couls** [comment]
eyeball seems really OP if it decides to team up with security, can end cult shifts very quickly if they can peer into every cult maint base they can make

**Comment by IndieanaJones** [comment]
> 
> 
> eyeball seems really OP if it decides to team up with security, can end cult shifts very quickly if they can peer into every cult maint base they can make

You can punch the eye to kill it pretty quickly, it's not strong.  It takes 15 damage from harm intent, and a cult could always send a wraith or the like to punch it to death before sec ever uses it.  Also, if the xenobiologist isn't on your side and making sentient mobs, it might be helpful to have a precursor warning for when he sends the entire army at you while you're trying summon Nar'Sie.

**Comment by Repukan** [comment]
Remove the magicarps

**Comment by Arathianen** [comment]
I came from my absence to leave a quick note:

A watcher should be left in. It has a VERY important function. If slaughtered, it drops diamonds. In very bad situations, I have slaughtered watchers to make diamonds when I really needed some. It's a niche but important use.

**Comment by IndieanaJones** [comment]
> I came from my absence to leave a quick note:
> 
> A watcher should be left in. It has a VERY important function. If slaughtered, it drops diamonds. In very bad situations, I have slaughtered watchers to make diamonds when I really needed some. It's a niche but important use.

Basilisks also drop diamonds, so we should still be good.

**Comment by Jessen21** [comment]
> I came from my absence to leave a quick note:
> A watcher should be left in. It has a VERY important function. If slaughtered, it drops diamonds. In very bad situations, I have slaughtered watchers to make diamonds when I really needed some. It's a niche but important use.
> 
> Basilisks also drop diamonds, so we should still be good.

thats not the ONLY thing they dropped tho. they also dropped bone, and that was kind of important too in some cases where miners aren't bringing any bone back.

**Comment by coiax** [comment]
The eyeball mob is OP sure, because it has xray vision, but it requires both light pink slimes and gold slimes to have been bred, which takes time. I'm no pro at Xenobiology, but I'd be interested to hear how quickly some of the real experts could get those slimes.

Also, Genetics has xray, don't they

**Comment by Tlaltecuhtli** [comment]
does this mess with life reaction?

**Comment by Ty-the-Smonk** [comment]
Give me evil crab back

**Comment by IndieanaJones** [comment]
I am a merciful god.  I have returned your evil crabs to you, but they have been banished to the water reaction, since despite being evil, they are completey harmless.

File: code/modules/mob/living/simple_animal/hostile/hivebot.dm [hunk 1]
```diff
@@ -93,7 +96,67 @@
...
+		if(fixable.health >= fixable.maxHealth)
+			to_chat(src, "<span class='warning'>Diagnostics indicate that this unit is at peak integrity.</span>")
+			return
+		to_chat(src, "<span class='warning'>You begin repairs...</span>")
+		if(do_after(src, 50, target = fixable))
+			fixable.revive(full_heal = TRUE, admin_revive = TRUE)
+			do_sparks(3, TRUE, fixable)
+			to_chat(src, "<span class='warning'>Repairs complete.</span>")
+		return
+	. = ..()
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/hivebot.dm]: return ..()

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 2]
```diff
@@ -273,3 +269,104 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
...
+/mob/living/simple_animal/hostile/mimic/xenobio/Initialize()
+	. = ..()
+	L = new
+	L.Grant(src)
+	
+/mob/living/simple_animal/hostile/mimic/xenobio/AttackingTarget()
+	if(src == target)
+		toggle_open()
+		return
+	. = ..()
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: return ..()

**Comment by Jessen21** [comment]
> 
> 
> I am a merciful god. I have returned your evil crabs to you, but they have been banished to the water reaction, since despite being evil, they are completey harmless. Also added cooldowns on the goldgrub's burrow and creature's phasing.

could you also PLEASE keep the normal watcher in so that Xenobio has some options for providing watcher sinew and bone? I know basilisks might have bone but I'm really worried about the watcher sinew vanishing, because thats something I like getting.

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 3]
```diff
@@ -273,3 +269,104 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
...
+	if(M.a_intent != "help")
+		return
+	toggle_open()
+			
+/mob/living/simple_animal/hostile/mimic/xenobio/death()
+	var/obj/structure/closet/crate/C = new(get_turf(src))
+	// Put loot in crate
+	for(var/atom/movable/AM in src)
+		AM.forceMove(C)
+	..()
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: return ..()

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goldgrub.dm [hunk 4]
```diff
@@ -59,16 +132,23 @@
...
-		if(length(loot) < 10)
-			var/using = min(10 - length(loot), O.amount)
-			for(var/i in 1 to using)
-				loot += O.type
-			O.use(using)
-	visible_message("<span class='notice'>The ore was swallowed whole!</span>")
-
+/mob/living/simple_animal/hostile/asteroid/goldgrub/death(gibbed)
+	barf_contents()
+	..()
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/goldgrub.dm]: return ..()

**Comment by Shadowflame909** [comment]
Just go mine then

Not everything has to be available for Xenobio

**Comment by IndieanaJones** [comment]
> has this been tested

Not on live, but I've tested every single mechanic on my own test server.

**Comment by AnonymousNow** [comment]
While you're here, could you tweak the lightgeist so that their "fixing" ability stops sitting on the fence?
If I remember correctly, they "fix" anything they interact with, which means healing mobs (it might not currently apply to all living mobtypes, though, that might be worth checking), but not actually fixing inanimate things that are damaged somehow. If fixing said objects could be implemented, or the language used when interacting with objects is itself fixed to remove references to "fixing" things, replacing it with "healing" things, that'd be much appreciated. Thank you.

**Comment by ATH1909** [comment]
Do goldgrubs have the ability to mine rock under their own power? Do they have the ability to see ores in walls (as if they had a mining scanner)?

**Comment by carlarctg** [comment]
dont add watchers in, bone and sinew are a reward for killing fauna not for sitting around in xenobio ranching slimed for 40 minutes

**Comment by IndieanaJones** [comment]
> While you're here, could you tweak the lightgeist so that their "fixing" ability stops sitting on the fence?
> If I remember correctly, they "fix" anything they interact with, which means healing mobs (it might not currently apply to all living mobtypes, though, that might be worth checking), but not actually fixing inanimate things that are damaged somehow. If fixing said objects could be implemented, or the language used when interacting with objects is itself fixed to remove references to "fixing" things, replacing it with "healing" things, that'd be much appreciated. Thank you.

Sure, I can do that.  In doing this, I've also fixed the issue where fugu would remove the lightgeist's healing ability, and tied the lightgeist's healing power into their harm damage value, so things like guu will boost the lightgeist's healing ability.

> Do goldgrubs have the ability to mine rock under their own power? Do they have the ability to see ores in walls (as if they had a mining scanner)?

They can mine rock on their accord (along with basilisks), but cannot see ores.  This balances itself though, since goldgrubs and basilisks can actually mine pretty damn fast.

>dont add watchers in, bone and sinew are a reward for killing fauna not for sitting around in xenobio ranching slimed for 40 minutes

No intent on adding them back in.

**Comment by IndieanaJones** [comment]
Added information regarding the lightgeist changes in the PR description.

**Comment by AnonymousNow** [comment]
> 
> 
> > While you're here, could you tweak the lightgeist so that their "fixing" ability stops sitting on the fence?
> > If I remember correctly, they "fix" anything they interact with, which means healing mobs (it might not currently apply to all living mobtypes, though, that might be worth checking), but not actually fixing inanimate things that are damaged somehow. If fixing said objects could be implemented, or the language used when interacting with objects is itself fixed to remove references to "fixing" things, replacing it with "healing" things, that'd be much appreciated. Thank you.
> 
> Sure, I can do that. In doing this, I've also fixed the issue where fugu would remove the lightgeist's healing ability, and tied the lightgeist's healing power into their harm damage value, so things like guu will boost the lightgeist's healing ability.
> 

Thank you. It bugged me for the longest time that you could, say, click on a cracked window or damaged mech, the text would tell you you're fixing it, but nothing would actually happen. I guess it was some sort of oversight from the past.

**Comment by IndieanaJones** [comment]
Updated to contain the borg spin nerf so it also applies to animals using the same tactic as well.

**Comment by Jessen21** [comment]
> 
> 
> > dont add watchers in, bone and sinew are a reward for killing fauna not for sitting around in xenobio ranching slimed for 40 minutes
> 
> No intent on adding them back in.

then add a new way to get sinew at the least. Bone could probably be picked up from basilisks, but I at least want SOME way to get sinew, especially if mining is incompetant. Please, I always try to have some on hand.

**Comment by Grazyn** [comment]
@Jessen21 There's a public shuttle that goes to lavaland at arrivals. Just go and kill them yourself. It isn't the miners' job to bring back that stuff unless you ask.

**Comment by Jessen21** [comment]
> 
> 
> @Jessen21 There's a public shuttle that goes to lavaland at arrivals. Just go and kill them yourself. It isn't the miners' job to bring back that stuff unless you ask.

well that last part depends on if there even IS mining to ask, a lot of the miners i've been dealing with are usually dead by the time i'm asking them for something. Not to mention actually FINDING watchers is a bit difficult.

**Comment by KomradeSpectre** [comment]
@IndieanaJones Did you uh...test this?

**Comment by AnonymousNow** [comment]
Just reconfirming my anticipation for this. Looking forward to testing geists on a ground level.

**Comment by dee-dubya** [comment]
For the riding bears feature, can you mount other people to you as a bear, like a borg currently can?

**Comment by IndieanaJones** [comment]
> For the riding bears feature, can you mount other people to you as a bear, like a borg currently can?

Yes.

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 5]
```diff
@@ -273,3 +269,104 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
...
+	attack_verb_continuous = "bites"
+	attack_verb_simple = "bite"
+	speak_emote = list("clatters")
+	gold_core_spawnable = HOSTILE_SPAWN
+	var/opened = FALSE
+	var/open_sound = 'sound/machines/crate_open.ogg'
+	var/close_sound = 'sound/machines/crate_close.ogg'
+	var/max_mob_size = MOB_SIZE_HUMAN
+	var/locked = FALSE
+	var/datum/action/innate/mimic/lock/L
```
Comment (kriskog) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: Single letter var

File: code/modules/mob/living/simple_animal/hostile/mimic.dm [hunk 6]
```diff
@@ -273,3 +269,104 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
...
+		else if(isitem(AM) && !HAS_TRAIT(AM, TRAIT_NODROP))
+			return TRUE
+	else
+		return FALSE
+	return TRUE
+	
+/datum/action/innate/mimic
+	background_icon_state = "bg_default"
+	
+/datum/action/innate/mimic/lock
```
Comment (kriskog) [inline code/modules/mob/living/simple_animal/hostile/mimic.dm]: Slap an open crate on a button, and you got a codersprite. Up to you.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm [hunk 7]
```diff
@@ -57,6 +67,27 @@
...
 		if(3)
 			adjustBruteLoss(110)
+			
+/mob/living/simple_animal/hostile/asteroid/basilisk/AttackingTarget()
+	. = ..()
+	if(lava_drinker && !warmed_up && istype(target, /turf/open/lava))
+		visible_message("<span class='warning'>[src] begins to drink from [target]...</span>")
+		if(do_after(src, 70, target = target))
+			visible_message("<span class='warning'>[src] begins to fire up!</span>")
+			revive(full_heal = TRUE, admin_revive = TRUE)
```
Comment (kriskog) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/basilisk.dm]: fully_heal()

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/goldgrub.dm [hunk 8]
```diff
@@ -25,26 +25,99 @@
...
 	status_flags = CANPUSH
+	gold_core_spawnable = HOSTILE_SPAWN
 	search_objects = 1
 	wanted_objects = list(/obj/item/stack/ore/diamond, /obj/item/stack/ore/gold, /obj/item/stack/ore/silver,
 						  /obj/item/stack/ore/uranium)
 
 	var/chase_time = 100
 	var/will_burrow = TRUE
+	var/datum/action/innate/goldgrub/spitore/S
+	var/datum/action/innate/goldgrub/burrow/B
```
Comment (kriskog) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/goldgrub.dm]: single letter vars

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm [hunk 9]
```diff
@@ -17,11 +17,99 @@
 	gold_core_spawnable = HOSTILE_SPAWN
 	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
 	minbodytemp = 0
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+	var/phaser = TRUE
+	var/datum/action/innate/creature/teleport/T
```
Comment (kriskog) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Single letter var

**Review by kriskog — CHANGES_REQUESTED** [review]
Mostly nitpicks, there are a few more spriteless action icons too but those are entirely up to you.

File: code/modules/mob/living/simple_animal/hostile/eyeballs.dm [hunk 10]
```diff
@@ -1,9 +1,17 @@
...
+	icon = 'icons/mob/carp.dmi'
 	icon_state = "eyeball"
 	icon_living = "eyeball"
 	icon_gib = ""
+	maxHealth = 45
+	health = 45
+	harm_intent_damage = 15
+	obj_damage = 60
+	melee_damage_lower = 20
+	melee_damage_upper = 25
```
[the remark this answers is not in the document]
Comment (IndieanaJones) [reply code/modules/mob/living/simple_animal/hostile/eyeballs.dm]: These are the original values the eyeball had, I didn't tinker with them.  I'm going to tweak them down to balance their new ability.

**Comment by IndieanaJones** [comment]
I'd rather have someone actually make some nice sprites for the abilities if they so choose to do so rather than make some ugly placebos for them instead.  They should be fine for now, since hovering your mouse over the ability still tells you what you need to know about it.

**Review by kriskog — APPROVED** [review]

**Pull request merged**